conbench/app/compare.py

```diff
@@ -272,8 +274,11 @@ def _compare(
         # Mutate comparison objs (dictionaries) on the fly
         for c in comparisons:
             view = "app.compare-benchmark-results"
-            compare = f'{c["baseline"]["benchmark_result_id"]}...{c["contender"]["benchmark_result_id"]}'
-            c["compare_benchmarks_url"] = f.url_for(view, compare_ids=compare)
+            if c["baseline"] and c["contender"]:
+                compare = f'{c["baseline"]["benchmark_result_id"]}...{c["contender"]["benchmark_result_id"]}'
+                c["compare_benchmarks_url"] = f.url_for(view, compare_ids=compare)
+            else:
+                c["compare_benchmarks_url"] = None
```

Comment:
Small bugfixes due to new API structure.

Reply:
Just added a test for this file's bugfixes (confirmed that it fails on `main` but passes now)

Reply:
Thanks for adding tests to this — we should have had them already, but better late than never!